client/src/utils/actions.test.js

```diff
@@ -0,0 +1,38 @@
+import { createAction } from './actions';
+
+describe('createAction', () => {
+  it('should return a function that creates an action that returns an object', () => {
+    expect(createAction()()).toEqual({});
```

Comment:
ok I will remove this test for now, this is the current behaviour and I think that changing it should not be in scope of this PR.

I do not know enough about this code and I am pretty sure that Redux will not be happy if we accidentally let non-objects leak into the actions.

From my knowledge an empty object is ok though, it will just do nothing.

However, I will remove this unit test and merge this PR in.

Reply:
👍 all I meant was switching to:

```
expect(createAction('SOME_ACTION_NAME')()).toEqual({ type: 'SOME_ACTION_NAME' });
```

so we technically follow the signature of the function. So it’s only the tests that need changing. Shouldn’t we be writing those tests in TypeScript as well? This way those issues would become apparent from the get-go.

Reply:
Ok all good - I'll leave this as is though - we can add more tests later if needed. 

My understanding is that we write tests in JavaScript - not TypeScript - I assumed that by convention and also added to the work in progress UI development docs. 

But I'm not sure whether I actually heard that anywhere. 

I'm personally a fan of JS unit tests as it allows you to write things that 'should not happen' in TS - remembering we want to ensure some code is more robust than just our types will allow. 

I have also historically found TS tests add a lot of overhead to write with little benefits in other projects. 

I don't think we have any unit tests in TS right now - but if that's something we want to do we should maybe add to the UI team agenda.

Reply:
I've also told other contributors not to write unit tests in TypeScript (either in some if the Outreachy issues or on PRs). But happy to align with whatever we want to do in the project - would be good to document the preferred approach once we all agree.